mosaic/queries/StatusdurationQuery.py

```diff
@@ -1,26 +1,33 @@
-import datetime
-
 from .BaseQuery import BaseQuery
 from .utils import date_difference
 
 
 class Transition():
-    def __init__(self, status, timestamp):
-        self.status = status
+    def __init__(self, to_status, from_status, timestamp):
+        self.to_status = to_status
+        self.from_status = from_status
         self.timestamp = timestamp
 
 
 class StatusdurationQuery(BaseQuery):
     supports_rolling = True
+
+    # This means you can use --rolling in combination with --end-date
+    supports_isolated_rolling = True
+
     query_bases = {
-        'statusduration': ('PROJECT = {project} '
-                           'AND TYPE IN ({types}) '
-                           'AND statusCategory = Done '
-                           'AND status CHANGED TO {end_state} '
-                           'DURING("{begin_date}", "{end_date}") '),
-        'statusduration_rolling': ('PROJECT = {project} '
-                                   'AND TYPE IN ({types}) '
-                                   'AND statusCategory = "In Progress" ')
+        'statusduration': (
+            'PROJECT = {project} '
+            'AND TYPE IN ({types}) '
+            'AND statusCategory = Done '
+            'AND status CHANGED TO {end_state} '
+            'DURING("{begin_date}", "{end_date}") '),
+        'statusduration_rolling': (
+            'PROJECT = {project} '
+            'AND TYPE IN ({types}) '
+            'AND status CHANGED TO "{argument}" BEFORE "{end_date}" '
```

Comment:
This line is the only one that is confusing to me. What's the reason for the "status changed to {argument}"?

Reply:
Ah, so if you're measuring "code review" duration here, you want to find cards that moved into code review sometimes before the end of the zone your measuring.  Here, `"{argument}"` is "Code Review".

Reply:
Ok, yep, it's been so long that I've messed with this that I totally forgot how this all worked! Makes sense.